python-threatexchange/threatexchange/exchanges/signal_exchange_api.py

```diff
@@ -63,6 +63,38 @@ class SignalExchangeAPI(
 
     """
 
+    @classmethod
+    @abstractmethod
+    def for_collab(cls, collab: TCollabConfig) -> "SignalExchangeAPI":
+        """
```

Comment:
I am fully behind this change!

I'm slightly concerned that there will now need to be fallbacks for credentials. eg. Does my collab config have credentials? Yes, use them. No, does my API have credentials?

But, I can think of ways around that in HMA. 

How are you planning on storing credentials in threatexchange-cli after this change goes in? Separately for each collab config?

Reply:
Chatted offline, conclusions:
1. Let's go ahead and refactor to for_collab(cls, collab)
2. Explicitly rejected (for now) `for_collab(cls, collab, only_signals)` or `for_collab(cls, collab, auth=None)`, though solving reflection-based auth is something @schatten is interested in solving for HMA, and so might be a 1.1.x feature.